### ISSUE TITLE
Specified port in config not used when using SSH

### ISSUE THREAD

**Issue by mpemberton5** [microsoft/vscode-remote-release#3591]
- VSCode Version: 1.48.2
- Local OS Version: Windows_NT x64 10.0.10240
- Remote OS Version: Ubuntu 18.04
- Remote Extension/Connection Type: SSH

Steps to Reproduce:

1. Added Remote Extension
2. Installed Git for Windows
3. Specified port in config file.  Remote does not use port 22 SSH

**Config file:**
Host remotehost.com
  HostName remotehost.com
  Port 14530
  User username

**Logs:**
[11:23:03.111] Log Level: 2
[11:23:03.120] remote-ssh@0.51.0
[11:23:03.120] win32 x64
[11:23:03.125] SSH Resolver called for "ssh-remote+remotehost.com", attempt 1
[11:23:03.125] SSH Resolver called for host: remotehost.com
[11:23:03.125] Setting up SSH remote "remotehost.com"
[11:23:03.217] Using commit id "a0111111111111111fa657e454193f72aef85bd0" and quality "stable" for server
[11:23:03.219] Install and start server if needed
[11:23:05.040] Checking ssh with "ssh -V"
[11:23:05.047] Got error from ssh: spawn ssh ENOENT
[11:23:05.048] Checking ssh with "C:\WINDOWS\System32\OpenSSH\ssh.exe -V"
[11:23:05.049] Got error from ssh: spawn C:\WINDOWS\System32\OpenSSH\ssh.exe ENOENT
[11:23:05.050] Checking ssh with "C:\Program Files\Git\usr\bin\ssh.exe -V"
[11:23:05.876] > OpenSSH_8.3p1, OpenSSL 1.1.1g  21 Apr 2020
[11:23:05.879] Running script with connection command: "C:\Program Files\Git\usr\bin\ssh.exe" -T -D 56573 remotehost.com bash
[11:23:05.884] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[11:23:06.250] > 
[11:23:06.250] Got some output, clearing connection timeout
[11:23:27.501] > ssh: connect to host remotehost.com port 22: Connection timed out
> The process tried to write to a nonexistent pipe.
> 
[11:23:27.778] "install" terminal command done
[11:23:27.779] Install terminal quit with output: The process tried to write to a nonexistent pipe.
[11:23:27.779] Received install output: The process tried to write to a nonexistent pipe.
[11:23:27.781] Resolver error: The process tried to write to a nonexistent pipe
[11:23:27.788] ------

**Comment by roblourens** [comment]
Sorry for the slow response. If you set the Host to be different from the HostName in your config file, does it work? Can you connect from an external terminal?

**Comment by mpemberton5** [comment]
Tried Setting "Host" to **RemoteServer**, and it tried to connect to "RemoteServer".  It did not use HostName or Port.  I tried some other scenario's, like changing case, removing the HostName option, etc, but it would never pull the "Port" config setting.

**Config file:**
Host RemoteServer
HostName remotehost.com
Port 14530
User username

**Logs:**
[10:05:56.553] Log Level: 2
[10:05:56.564] remote-ssh@0.55.0
[10:05:56.564] win32 x64
[10:05:56.568] SSH Resolver called for "ssh-remote+7b22686f73744e616d65223a2252656d6f7465536572766572227d", attempt 1
[10:05:56.569] SSH Resolver called for host: RemoteServer
[10:05:56.569] Setting up SSH remote "RemoteServer"
[10:05:56.604] Using commit id "2af051012b66169dde0c4dfae3f5ef48f787ff69" and quality "stable" for server
[10:05:56.607] Install and start server if needed
[10:05:59.049] Checking ssh with "ssh -V"
[10:05:59.056] Got error from ssh: spawn ssh ENOENT
[10:05:59.057] Checking ssh with "C:\WINDOWS\System32\OpenSSH\ssh.exe -V"
[10:05:59.059] Got error from ssh: spawn C:\WINDOWS\System32\OpenSSH\ssh.exe ENOENT
[10:05:59.060] Checking ssh with "C:\Program Files\Git\usr\bin\ssh.exe -V"
[10:05:59.198] > OpenSSH_8.3p1, OpenSSL 1.1.1g  21 Apr 2020

[10:05:59.208] Running script with connection command: "C:\Program Files\Git\usr\bin\ssh.exe" -T -D 65223 RemoteServer bash
[10:05:59.215] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[10:05:59.457] > 
[10:05:59.458] Got some output, clearing connection timeout
[10:06:01.886] > ssh: Could not resolve hostname remoteserver: Name or service not known
> The process tried to write to a nonexistent pipe.
> 
[10:06:02.161] "install" terminal command done
[10:06:02.162] Install terminal quit with output: The process tried to write to a nonexistent pipe.
[10:06:02.162] Received install output: The process tried to write to a nonexistent pipe.
[10:06:02.172] Resolver error: Error: The process tried to write to a nonexistent pipe
	at Function.Offline (c:\Users\mpembert\.vscode\extensions\ms-vscode-remote.remote-ssh-0.55.0\out\extension.js:1:130625)
	at c:\Users\mpembert\.vscode\extensions\ms-vscode-remote.remote-ssh-0.55.0\out\extension.js:1:126609
	at Object.t.handleInstallOutput (c:\Users\mpembert\.vscode\extensions\ms-vscode-remote.remote-ssh-0.55.0\out\extension.js:1:126987)
	at I (c:\Users\mpembert\.vscode\extensions\ms-vscode-remote.remote-ssh-0.55.0\out\extension.js:127:106775)
	at processTicksAndRejections (internal/process/task_queues.js:94:5)
	at async c:\Users\mpembert\.vscode\extensions\ms-vscode-remote.remote-ssh-0.55.0\out\extension.js:127:104774
	at async Object.t.withShowDetailsEvent (c:\Users\mpembert\.vscode\extensions\ms-vscode-remote.remote-ssh-0.55.0\out\extension.js:127:109845)
	at async Object.t.resolve (c:\Users\mpembert\.vscode\extensions\ms-vscode-remote.remote-ssh-0.55.0\out\extension.js:127:107960)
	at async c:\Users\mpembert\.vscode\extensions\ms-vscode-remote.remote-ssh-0.55.0\out\extension.js:127:141955
[10:06:02.179] ------

**Comment by roblourens** [comment]
It seems like you are using Git ssh but it is not using your ssh config file for some reason.

Are you able to connect from an external terminal?

**Comment by mpemberton5** [comment]
Ahh, you were spot on.  I launched Git Bash and first noticed I defaulted to my \h\ drive.  Being a work computer, I didn't think much of it, but since that must be my home drive, SSH was looking under there for my config file.  I confirmed that with the `-vvv` flag.  I simply went into settings and set the custom config file path which is on my C drive and worked like a dream.  So would the code need to look for the default home drive (in case it isn't C:)?  Just a thought.  Maybe this will help anyone else with the same/similar issue.

I'll close but feel free to open it back up if you feel there might be code that could be adjusted.

Thanks for your help!
